From 8700efc7ff12e0ebc99ffc12afcf4a1156973534 Mon Sep 17 00:00:00 2001
From: Garrett Brown <garrett.brown@aclima.earth>
Date: Wed, 19 Aug 2026 18:54:48 -0700
Subject: [PATCH] Fix ejecting disks

---
 libretro.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/libretro.c b/libretro.c
index 28933f3c..a55c14e3 100644
--- a/libretro.c
+++ b/libretro.c
@@ -3866,6 +3866,16 @@ static bool disk_set_image_index(unsigned index)
     * than letting them flow through to CD_SelectedDisc--. */
    if (num_images == 0)
       return false;
+
+   /* An index of num_images is a request to eject the disc in the libretro API. */
+   if (index == num_images)
+   {
+      eject_state = true;
+      CD_SelectedDisc = -1;
+      DoSimpleCommand(MDFN_MSC_EJECT_DISK);
+      return true;
+   }
+
    if (index >= num_images)
       index = num_images - 1;
 
-- 
2.43.0

