From 7051cf83f8d0162a6a208b435c496f2f4fc7eb00 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 May 2026 17:39:01 -0700
Subject: [PATCH 2/2] Fix PSX CPU ICache assertion after C conversion

---
 mednafen/psx/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mednafen/psx/cpu.c b/mednafen/psx/cpu.c
index d16483fc..a341d7f6 100644
--- a/mednafen/psx/cpu.c
+++ b/mednafen/psx/cpu.c
@@ -288,7 +288,7 @@ void CPU_Power(PS_CPU *self)
    unsigned i;
 
    (void)self;
-   assert(sizeof(s_cpu.ic.ICache) == sizeof(s_cpu.ic.ICache_Bulk));
+   assert(sizeof(ICache) == sizeof(ICache_Bulk));
 
    memset(GPR, 0, 32 * sizeof(uint32_t));
    memset(&cpu_CP0, 0, sizeof(cpu_CP0));
-- 
2.50.1 (Apple Git-155)

