From 1a853f937d027f799516fa522a2f7eadc82c545f Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 May 2026 17:44:53 -0700
Subject: [PATCH 3/4] Fix macOS Clang build failures after C conversion

---
 rsx/rsx_intf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rsx/rsx_intf.c b/rsx/rsx_intf.c
index e5328019..0b06ac71 100644
--- a/rsx/rsx_intf.c
+++ b/rsx/rsx_intf.c
@@ -125,7 +125,7 @@ enum rsx_renderer_type rsx_intf_is_type(void)
    return rsx_type;
 }
 
-inline void rsx_intf_dump_init(void)
+static void rsx_intf_dump_init(void)
 {
 #if defined(RSX_DUMP)
    {
-- 
2.50.1 (Apple Git-155)

