From 5e773407eda6aad398f101939654f987d8c2a088 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 6 May 2026 17:45:11 -0700
Subject: [PATCH 4/4] Fix Android build after C conversion

---
 Makefile.common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.common b/Makefile.common
index e232fe76..4891fa39 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -178,7 +178,7 @@ ifeq ($(HAVE_LIGHTREC), 1)
                -include $(CORE_DIR)/lightning-lightrec-include/debug.h
 endif
 
-   SOURCES_CXX += $(CORE_EMU_DIR)/cpu.cpp
+   SOURCES_CXX += $(CORE_EMU_DIR)/cpu.c
 
    ifeq ($(HAVE_JIT), 1)
       SOURCES_CXX += $(CORE_EMU_DIR)/decomp.cpp
-- 
2.50.1 (Apple Git-155)

